projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/JobImageBuilder.java

```diff
@@ -155,14 +153,42 @@ public boolean buildImage(
       return false;
     }
 
+    String awsRegion;
+    String builderAwsSecretAccessKey;
+    String builderAwsAccessKeyId;
+    String builderAwsSessionToken;
+
     // Rebuild the image if the Python version changes but the gitCommitSha remains the same.
-    if ((actualDataJobDeployment == null
-            || desiredDataJobDeployment
-                .getPythonVersion()
-                .equals(actualDataJobDeployment.getPythonVersion()))
-        && dockerRegistryService.dataJobImageExists(imageName, credentials)) {
-      log.trace("Data Job image {} already exists and nothing else to do.", imageName);
-      return true;
+    if (registryType.equalsIgnoreCase(REGISTRY_TYPE_ECR)) {
+      var credentials = awsCredentialsService.createTemporaryCredentials();
+      builderAwsSecretAccessKey = credentials.awsSecretAccessKey();
+      builderAwsAccessKeyId = credentials.awsAccessKeyId();
+      builderAwsSessionToken = credentials.awsSessionToken();
+      awsRegion = credentials.region();
+      if ((actualDataJobDeployment == null
+              || desiredDataJobDeployment
+                  .getPythonVersion()
+                  .equals(actualDataJobDeployment.getPythonVersion()))
+          && ecrRegistryInterface.checkEcrImageExists(imageName, credentials)) {
+        log.trace("Data Job image {} already exists and nothing else to do.", imageName);
+        return true;
+      }
+    } else {
+      builderAwsSecretAccessKey = REGISTRY_TYPE_GENERIC;
```

Comment:
Why is the generic not empty?

Reply:
I would rather have a specific value in these variables, which are later on passed to the builders.